eth/client.go

```diff
@@ -150,306 +226,601 @@ func NewTransactOptsForAccount(account accounts.Account, passphrase string, keyS
+
+		receipt, err := c.WaitForReceipt(tx)
+		if err != nil {
+			outErr <- err
+		} else {
+			outRes <- *receipt
+		}
+
+		return
+	}()
+
+	return outRes, outErr
 }
 
-func (c *Client) TranscoderStake() (*big.Int, error) {
-	return c.protocolSession.TranscoderTotalStake(c.account.Address)
+func (c *Client) Transfer(toAddr common.Address, amount *big.Int) (<-chan types.Receipt, <-chan error) {
+	outRes := make(chan types.Receipt)
+	outErr := make(chan error)
+
+	go func() {
+		defer close(outRes)
+		defer close(outErr)
+
+		tx, err := c.tokenSession.Transfer(toAddr, amount)
+		if err != nil {
+			outErr <- err
+			return
+		}
+
+		glog.Infof("[%v] Submitted transaction %v. Transfer %v LPTU to %v", c.account.Address.Hex(), tx.Hash().Hex(), amount, toAddr.Hex())
+
+		receipt, err := c.WaitForReceipt(tx)
+		if err != nil {
+			outErr <- err
+		} else {
+			outRes <- *receipt
+		}
+
+		return
+	}()
+
+	return outRes, outErr
 }
 
-func (c *Client) Bond(amount *big.Int, toAddr common.Address) (*types.Transaction, error) {
-	var (
-		logsCh = make(chan types.Log)
-	)
+func (c *Client) Approve(toAddr common.Address, amount *big.Int) (chan types.Log, chan error) {
+	outRes := make(chan types.Log)
+	outErr := make(chan error)
+
+	go func() {
+		defer close(outRes)
+		defer close(outErr)
+
+		logsCh, sub, err := c.SubscribeToApproval()
+
+		defer close(logsCh)
+		defer sub.Unsubscribe()
+
+		if err != nil {
+			outErr <- err
+			return
+		}
+
+		_, err = c.tokenSession.Approve(toAddr, amount)
+		if err != nil {
+			outErr <- err
+			return
+		}
+
+		log := <-logsCh
+
+		outRes <- log
 
-	tokenJson, err := abi.JSON(strings.NewReader(contracts.LivepeerTokenABI))
+		return
+	}()
 
+	return outRes, outErr
+}
+
+func (c *Client) SubscribeToApproval() (chan types.Log, ethereum.Subscription, error) {
+	logsCh := make(chan types.Log)
+
+	abiJSON, err := abi.JSON(strings.NewReader(contracts.LivepeerTokenABI))
 	if err != nil {
-		glog.Errorf("Error decoding ABI into JSON: %v", err)
-		return nil, err
+		return nil, nil, err
 	}
 
 	q := ethereum.FilterQuery{
 		Addresses: []common.Address{c.tokenAddr},
-		Topics:    [][]common.Hash{[]common.Hash{tokenJson.Events["Approval"].Id()}, []common.Hash{common.BytesToHash(common.LeftPadBytes(c.account.Address[:], 32))}},
+		Topics:    [][]common.Hash{[]common.Hash{abiJSON.Events["Approval"].Id()}, []common.Hash{common.BytesToHash(common.LeftPadBytes(c.account.Address[:], 32))}},
 	}
 
 	ctx, _ := context.WithTimeout(context.Background(), c.rpcTimeout)
 
-	logsSub, err := c.backend.SubscribeFilterLogs(ctx, q, logsCh)
+	sub, err := c.backend.SubscribeFilterLogs(ctx, q, logsCh)
+	if err != nil {
+		return nil, nil, err
+	}
 
-	defer logsSub.Unsubscribe()
-	defer close(logsCh)
+	return logsCh, sub, nil
+}
 
-	tx, err := c.tokenSession.Approve(c.protocolAddr, amount)
+// CONSTANT FUNCTIONS
 
+func (c *Client) RoundInfo() (*big.Int, *big.Int, *big.Int, error) {
```

Comment:
Could probably use a comment describing these return values, since they don't map directly to a protocol function.